Implement "Edit commit" by breaking after current commit

stefanhaller opened the pull request:
- **PR Description**

Instead of rebasing from the commit below the current one and then setting the
current one to "edit", we rebase from the current one and insert a "break" after
it. In most cases the behavior is exactly the same as before, except that the
new method also works if the current commit is a merge commit. This is useful if
you want to create a new commit at the very beginning of your branch (before the
first commit of your branch), if the last commit before that is a merge.
- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
   * I didn't add or adapt any tests, because for the normal cases the behavior should be unchanged
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

stefanhaller commented on pkg/app/app.go:
```diff
@@ -123,7 +123,7 @@ func (app *App) validateGitVersion() (*git_commands.GitVersion, error) {
 	}
 
-	if version.IsOlderThan(2, 0, 0) {
+	if version.IsOlderThan(2, 20, 0) {
```
@jesseduffield Not sure you'll agree with this. Personally I find it a bit insane to support nine-year-old git versions, so I'd have no problem with this change; but if you object, I think it should be possible to fall back to the old way of editing a commit if the git version is older than 2.20. Let me know what you prefer.

jesseduffield replied:
I'm fine with it. 2.0 was arbitrary in the first place

stefanhaller replied:
OK, great. I added a commit to revert a compatibility fix that was added for supporting versions older than 2.7, which we now no longer need; @Ryooooooga, please have a look.

Ryooooooga replied:
@stefanhaller :+1:

jesseduffield commented:
This is great, I'm all for it. We currently raise an error upon trying to start an interactive rebase from the first commit i.e.:
```go
	// pkg/commands/git_commands/rebase.go:165
	if len(commits) <= baseIndex {
		return nil, "", errors.New(self.Tr.CannotRebaseOntoFirstCommit)
	}
```
Is this check no longer required thanks to this PR? If so, would be good to remove that guard and add an integration test showing that this works from the first commit

stefanhaller commented:
> Is this check no longer required thanks to this PR? If so, would be good to remove that guard and add an integration test showing that this works from the first commit

Removing the guard is not possible, we still need it to guard against rewording the first commit, which is still not possible. (Side note: when rewording, you get the error message only _after_ having typed the new commit message, it would be nice if we would error out before that.)

But I added an integration test for editing the first commit, which is now indeed possible.

stefanhaller commented:
> But I added an integration test for editing the first commit, which is now indeed possible.

I take that back, this test doesn't belong here. The fact that this branch _also_ allows editing the first commit is a bit of a coincidence. I dropped the test again from here.

I created another PR which is independent from this one, which allows all interactive rebase operations down to the initial commit; I added the test there (and a few others): #2453

jesseduffield commented:
Tested it locally, all works fine. Thanks for making this!
